### PR TITLE
chore: drop Node polyfills for browser bundles

### PR DESCRIPTION
**Pull request by balazsorban44**
### What?

Drop unused libraries from Next.js:
```
assert 92K
browserify-zlib 72K
buffer 40K
constants-browserify 16K
crypto-browserify 368K    
domain-browser 12K
events 20K
https-browserify 16K
os-browserify 16K
path-browserify 20K
process 16K
punycode 12K
querystring-es3 12K
setimmediate 12K
stream-browserify 64K
stream-http 72K
string_decoder 20K
timers-browserify 12K
tty-browserify 16K
util 56K
vm-browserify 16K
```

### Why?

We are shipping some Node.js compatibility libraries for browsers, but they have not been used for a while, adding extra weight to Next.js

### How?

Drop the `experimental.fallbackNodePolyfills` option. Since nobody has ever commented on the new behavior of not shipping these libs to browsers, it should be safe to remove it. If a library has depended on these for some reason, the user could add their own polyfills instead of `next.config.js`, but likely the failing library should not be depending on Node.js APIs if it's expected to be used in the browser instead.

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites <!--17426666909-->
Commit: 48f91158a6fb19de2f2b8ab1acc5f421d712b7b1

`pnpm test test/integration/edge-runtime-response-error/test/index.test.js` 
- Edge runtime code with imports > test error if response is not Response type > Edge API dev test Response

<details>
<summary>Expand output</summary>

  ● Edge runtime code with imports › test error if response is not Response type › Edge API dev test Response

    expect(received).toContain(expected) // indexOf

    Expected substring: "Expected an instance of Response to be returned"
    Received string:    "
    Warning: Webpack is configured while Turbopack is not, which may cause problems.·
      See instructions if you need to configure Turbopack:
      https://turbo.build/pack/docs/features/customizing-turbopack·
      ⚠ middleware error - [project]/packages/next/dist/esm/build/polyfills/process.js - Error resolving commonjs request
    unable to resolve module \"next\" with subpath \"/dist/compiled/process\"·
     error - packages/next/dist/esm/build/polyfills/process.js:2:214  Error resolving commonjs request·
      1 | var _global_process, _global_process1;
    > 2 | module.exports = ((_global_process = global.process) == null ? void 0 : _global_process.env) && typeof ((_global_process1 = global.process) == null ? void 0 : _global_process1.env) === \"object\" ? global.process : require(\"next/dist/compiled/process\");
        |                                                                                                                                                                                                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      3 |
      4 | //# sourceMappingURL=process.js.map
    unable to resolve module \"next\" with subpath \"/dist/compiled/process\"
    It was not possible to find the requested file.
        Parsed request as written in source code: module \"next\" with subpath \"/dist/compiled/process\"
        Path where resolving has started: [project]/packages/next/dist/esm/build/polyfills/process.js
        Type of request: commonjs request
        Import map: aliased to module \"next\" with subpath \"/dist/compiled/process\" inside of [project]/test/integration/edge-runtime-response-error·······
      ⚠ /api/route error - [project]/packages/next/dist/esm/build/polyfills/process.js - Error resolving commonjs request
    unable to resolve module \"next\" with subpath \"/dist/compiled/process\"·
     error - packages/next/dist/esm/build/polyfills/process.js:2:214  Error resolving commonjs request·
      1 | var _global_process, _global_process1;
    > 2 | module.exports = ((_global_process = global.process) == null ? void 0 : _global_process.env) && typeof ((_global_process1 = global.process) == null ? void 0 : _global_process1.env) === \"object\" ? global.process : require(\"next/dist/compiled/process\");
        |                                                                                                                                                                                                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      3 |
      4 | //# sourceMappingURL=process.js.map
    unable to resolve module \"next\" with subpath \"/dist/compiled/process\"
    It was not possible to find the requested file.
        Parsed request as written in source code: module \"next\" with subpath \"/dist/compiled/process\"
        Path where resolving has started: [project]/packages/next/dist/esm/build/polyfills/process.js
        Type of request: commonjs request
        Import map: aliased to module \"next\" with subpath \"/dist/compiled/process\" inside of [project]/test/integration/edge-runtime-response-error·······
      ⚠ /api/route error - [project]/packages/next/dist/esm/shared/lib/isomorphic/path.js - Error resolving commonjs request
    unable to resolve module \"next\" with subpath \"/dist/compiled/path-browserify\"·
     error - packages/next/dist/esm/shared/lib/isomorphic/path.js:7:12  Error resolving commonjs request·
       5 |  */ let path;
       6 | if (process.env.NEXT_RUNTIME === \"edge\") {
    >  7 |     path = require(\"next/dist/compiled/path-browserify\");
         |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       8 | } else {
       9 |     path = require(\"path\");
      10 | }
    unable to resolve module \"next\" with subpath \"/dist/compiled/path-browserify\"
    It was not possible to find the requested file.
        Parsed request as written in source code: module \"next\" with subpath \"/dist/compiled/path-browserify\"
        Path where resolving has started: [project]/packages/next/dist/esm/shared/lib/isomorphic/path.js
        Type of request: commonjs request
        Import map: aliased to module \"next\" with subpath \"/dist/compiled/path-browserify\" inside of [project]/test/integration/edge-runtime-response-error·······
     ⨯ Error: Cannot find module 'next/dist/compiled/path-browserify'
        at <unknown> (file:///root/actions-runner/_work/next.js/next.js/test/integration/edge-runtime-response-error/.next/server/edge/chunks/edge ssr/edge-wrapper_9c09f4.js:491)

      at <unknown> (../file:/root/actions-runner/_work/next.js/next.js/test/integration/edge-runtime-response-error/.next/server/edge/chunks/edge ssr/[project]__63ab2b._.js:712:19)
      at Object.<anonymous> (../file:/root/actions-runner/_work/next.js/next.js/test/integration/edge-runtime-response-error/.next/server/edge/chunks/edge ssr/[project]__63ab2b._.js:715:7)
      at [project]/packages/next/dist/esm/shared/lib/isomorphic/path.js (ecmascript, edge ssr) (../file:/root/actions-runner/_work/next.js/next.js/test/integration/edge-runtime-response-error/.next/server/edge/chunks/edge ssr/[project]__63ab2b._.js:721:3)
      at <unknown> (../file:/root/actions-runner/_work/next.js/next.js/test/integration/edge-runtime-response-error/.next/server/edge/chunks/edge ssr/edge-wrapper_9c09f4.js:469:27)
      at runModuleExecutionHooks (../file:/root/actions-runner/_work/next.js/next.js/test/integration/edge-runtime-response-error/.next/server/edge/chunks/edge ssr/edge-wrapper_9c09f4.js:507:9)
      at instantiateModule (../file:/root/actions-runner/_work/next.js/next.js/test/integration/edge-runtime-response-error/.next/server/edge/chunks/edge ssr/edge-wrapper_9c09f4.js:468:9)
      at getOrInstantiateModuleFromParent (../file:/root/actions-runner/_work/next.js/next.js/test/integration/edge-runtime-response-error/.next/server/edge/chunks/edge ssr/edge-wrapper_9c09f4.js:539:12)
      at esmImport (../file:/root/actions-runner/_work/next.js/next.js/test/integration/edge-runtime-response-error/.next/server/edge/chunks/edge ssr/edge-wrapper_9c09f4.js:119:20)
      at <unknown> (../file:/root/actions-runner/_work/next.js/next.js/test/integration/edge-runtime-response-error/.next/server/edge/chunks/edge ssr/[project]__63ab2b._.js:1376:154)
      at [project]/packages/next/dist/esm/server/lib/incremental-cache/file-system-cache.js (ecmascript, edge ssr) (../file:/root/actions-runner/_work/next.js/next.js/test/integration/edge-runtime-response-error/.next/server/edge/chunks/edge ssr/[project]__63ab2b._.js:1653:3)
        ⚠ _app error - [project]/packages/next/dist/build/polyfills/process.js - Error resolving commonjs request
      unable to resolve module \"next\" with subpath \"/dist/compiled/process\"·
       error - packages/next/dist/build/polyfills/process.js:3:214  Error resolving commonjs request·
        1 | \"use strict\";
        2 | var _global_process, _global_process1;
      > 3 | module.exports = ((_global_process = global.process) == null ? void 0 : _global_process.env) && typeof ((_global_process1 = global.process) == null ? void 0 : _global_process1.env) === \"object\" ? global.process : require(\"next/dist/compiled/process\");
          |                                                                                                                                                                                                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        4 |
        5 | //# sourceMappingURL=process.js.map
      unable to resolve module \"next\" with subpath \"/dist/compiled/process\"
      It was not possible to find the requested file.
          Parsed request as written in source code: module \"next\" with subpath \"/dist/compiled/process\"
          Path where resolving has started: [project]/packages/next/dist/build/polyfills/process.js
          Type of request: commonjs request
          Import map: aliased to module \"next\" with subpath \"/dist/compiled/process\" inside of [project]/test/integration/edge-runtime-response-error·······
        ⚠ /_error error - [project]/packages/next/dist/build/polyfills/process.js - Error resolving commonjs request
      unable to resolve module \"next\" with subpath \"/dist/compiled/process\"·
       error - packages/next/dist/build/polyfills/process.js:3:214  Error resolving commonjs request·
        1 | \"use strict\";
        2 | var _global_process, _global_process1;
      > 3 | module.exports = ((_global_process = global.process) == null ? void 0 : _global_process.env) && typeof ((_global_process1 = global.process) == null ? void 0 : _global_process1.env) === \"object\" ? global.process : require(\"next/dist/compiled/process\");
          |                                                                                                                                                                                                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        4 |
        5 | //# sourceMappingURL=process.js.map
      unable to resolve module \"next\" with subpath \"/dist/compiled/process\"
      It was not possible to find the requested file.
          Parsed request as written in source code: module \"next\" with subpath \"/dist/compiled/process\"
          Path where resolving has started: [project]/packages/next/dist/build/polyfills/process.js
          Type of request: commonjs request
          Import map: aliased to module \"next\" with subpath \"/dist/compiled/process\" inside of [project]/test/integration/edge-runtime-response-error·······
      "
      at Object.toContain (integration/edge-runtime-response-error/test/index.test.js:70:35)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

File: packages/next/src/build/webpack-config.ts [hunk 1]
```diff
@@ -1806,91 +1806,30 @@ export default async function getBaseWebpackConfig(
...
-                          ),
-                          zlib: require.resolve(
-                            'next/dist/compiled/browserify-zlib'
-                          ),
-                          events: require.resolve('next/dist/compiled/events/'),
-                          setImmediate: require.resolve(
-                            'next/dist/compiled/setimmediate'
-                          ),
-                        },
+                  fallback: {
```
Comment (styfle) [inline packages/next/src/build/webpack-config.ts:1809]: I think you can delete this code now

> Webpack 5 no longer polyfills Node.js core modules automatically which means if you use them in your code running in browsers or alike, you will have to install compatible modules from npm and include them yourself. 

https://webpack.js.org/configuration/resolve/#resolvefallback

File: packages/next/src/compiled/@edge-runtime/cookies/index.js [hunk 2]
```diff
@@ -28,6 +28,87 @@ __export(src_exports, {
 });
 module.exports = __toCommonJS(src_exports);
 
+
+/**
```
Comment (styfle) [inline packages/next/src/compiled/@edge-runtime/cookies/index.js:32]: This code addition is unexpected 🤔

Comment (balazsorban44) [reply packages/next/src/compiled/@edge-runtime/cookies/index.js:32]: Will fix via https://github.com/vercel/next.js/pull/57082

**Comment by balazsorban44** [comment]
Unfortunately, we cannot do this yet because the default setting was `undefined`, not `false`, meaning this would be too disruptive without a notice.

**Pull request closed without merging**